tests/send-to-helix.proj

```diff
@@ -0,0 +1,101 @@
+    <_TestCoverageCommand Include="--settings $(_HelixCorrelationPayloadEnvVar)/support-data/CodeCoverage.config" />
+    <_TestCoverageCommand Include="--output $(_HelixLogsPath)/Tests.cobertura.xml" />
+
+    <_TestRunCommandArguments Include="dotnet test" />
+    <_TestRunCommandArguments Include="-s .runsettings" />
+    <_TestRunCommandArguments Include="$(_TestNameEnvVar).dll" />
+    <_TestRunCommandArguments Include="--ResultsDirectory:$(_HelixLogsPath)" />
+  </ItemGroup>
+
+  <Target Name="BuildHelixWorkItems" DependsOnTargets="_StageDependenciesForHelix">
+    <Error Condition="'$(RepoTestResultsPath)' == ''" Text="%24(RepoTestResultsPath) is unset" />
+
+    <PropertyGroup>
+      <HelixResultsDestinationDir>$(RepoTestResultsPath)</HelixResultsDestinationDir>
+
+      <_TestRunCommand Condition="'$(RunWithCodeCoverage)' == 'true'">@(_TestCoverageCommand, ' ') &quot;@(_TestRunCommandArguments, ' ')&quot;</_TestRunCommand>
+      <_TestRunCommand Condition="'$(RunWithCodeCoverage)' != 'true'">@(_TestRunCommandArguments, ' ')</_TestRunCommand>
+    </PropertyGroup>
+
+    <Error Condition="'$(_DotNetCoverageToolPath)' == '' or !Exists($(_DotNetCoverageToolPath))"
+           Text="Could not find dotnet-coverage tool. %24(_DotNetCoverageToolPath)=$(_DotNetCoverageToolPath)" />
+
+    <ItemGroup>
+      <_DefaultWorkItems Include="$(WorkItemArchiveWildCard)" />
+
+      <HelixWorkItem Include="@(_DefaultWorkItems -> '%(FileName)')">
+        <PayloadArchive>%(Identity)</PayloadArchive>
+        <PreCommands Condition="'$(OS)' == 'Windows_NT'">set &quot;TEST_NAME=%(FileName)&quot;</PreCommands>
+        <PreCommands Condition="'$(OS)' != 'Windows_NT'">export &quot;TEST_NAME=%(FileName)&quot;</PreCommands>
+        <Command>$(_TestRunCommand)</Command>
+        <Timeout>$(_workItemTimeout)</Timeout>
+
+        <!-- Download results file so coverage files can be extracted -->
+        <DownloadFilesFromResults>logs/Tests.cobertura.xml</DownloadFilesFromResults>
+      </HelixWorkItem>
+    </ItemGroup>
+
+    <Message Text="HelixCorrelationPayload: %(HelixCorrelationPayload.Identity)" Condition="'$(HelixDryRun)' == 'true'" Importance="High" />
+    <Message Text="HelixWorkItem: %(HelixWorkItem.Identity), Command: %(HelixWorkItem.Command), PreCommands: %(HelixWorkItem.PreCommands) with PayloadArchive: %(HelixWorkItem.PayloadArchive)" Condition="'$(HelixDryRun)' == 'true'" Importance="High" />
+
+    <Error Text="Stopping the build for dry run" Condition="'$(HelixDryRun)' == 'true'" />
+  </Target>
+
+  <Target Name="_StageDependenciesForHelix">
+    <PropertyGroup>
+      <_DotNetToolJsonPath>$(RepoRoot).config/dotnet-tools.json</_DotNetToolJsonPath>
+      <_DotNetToolJsonContent>$([System.IO.File]::ReadAllText($(_DotNetToolJsonPath)))</_DotNetToolJsonContent>
+      <_DotNetCoverageVersionRegex>(?:"dotnet-coverage": *{.*?"version": *")([^"]*?)(?:")</_DotNetCoverageVersionRegex>
```

Comment:
NIT: Not really important, but I think you can get rid of the first and third non capturing groups (meaning removing the parens and questionmark from (?:.......), and in theory you can also change the space matching from ` *` to `\s*` which would also match a new line. Didn't check but that might also mean you don't necessarily have to use Singleline. Anyway, not really important

Reply:
re:groups, that makes sense. I had that before I figured out how to get a group value out! I'll make the changes.